spec/services/candidate_interface/course_recommender_spec.rb

```diff
@@ -0,0 +1,442 @@
+        end
+      end
+
+      context 'when the Candidate has submitted several Application Choices' do
+        it "sets the 'subjects' parameter to a combination of all Subject codes" do
+          a1_subject = create(:subject, code: 'A1')
+          b2_subject = create(:subject, code: 'B2')
+          c1_subject = create(:subject, code: 'C1')
+          a1_course = create(:course, subjects: [a1_subject])
+          b2_course = create(:course, subjects: [b2_subject])
+          mixed_course = create(:course, subjects: [c1_subject, a1_subject, b2_subject])
+
+          a1_subject_course_option = create(:course_option, course: a1_course)
+          b2_subject_course_option = create(:course_option, course: b2_course)
+          mixed_subject_course_option = create(:course_option, course: mixed_course)
+
+          candidate = create(:candidate)
+          application_form = create(:application_form, candidate: candidate, application_choices: [])
+          _application_choices = [
+            create(:application_choice, :awaiting_provider_decision, application_form:, course_option: a1_subject_course_option),
+            create(:application_choice, :awaiting_provider_decision, application_form:, course_option: b2_subject_course_option),
+            create(:application_choice, :awaiting_provider_decision, application_form:, course_option: mixed_subject_course_option),
+          ]
+
+          uri = URI(described_class.recommended_courses_url(candidate:))
+          query_parameters = Rack::Utils.parse_query(uri.query)
+
+          expect(query_parameters['subjects[]']).to eq(%w[A1 B2 C1])
+        end
+      end
+    end
+
+    describe "the 'radius', 'latitude' and 'longitude' parameters" do
+      context 'when a Locatable is not specified' do
+        it 'does not set the any of the parameters' do
+          candidate = create(:candidate)
+
+          uri = URI(described_class.recommended_courses_url(candidate:, locatable: nil))
+          query_parameters = Rack::Utils.parse_query(uri.query)
+
+          expect(query_parameters).not_to have_key('radius')
+          expect(query_parameters).not_to have_key('latitude')
+          expect(query_parameters).not_to have_key('longitude')
+        end
+      end
+
+      context 'when a Locatable is specified' do
+        it 'sets the parameters to the Locatable values' do
+          locatable = instance_double(Provider, latitude: 51.5074, longitude: 0.1278)
+          candidate = create(:candidate)
+
+          uri = URI(described_class.recommended_courses_url(candidate:, locatable: locatable))
+          query_parameters = Rack::Utils.parse_query(uri.query)
+
+          expect(query_parameters['radius']).to eq('10')
+          expect(query_parameters['latitude']).to eq('51.5074')
+          expect(query_parameters['longitude']).to eq('0.1278')
+        end
+      end
+
+      context "when the Locatable doesn't have all the location data" do
```

Comment:
It still sets the other relevant parameters, right? Just none of the location related ones?

Reply:
I'm adding a "'mixture" context to make sure but yeah it still works 🎉